src/main/java/com/aws/greengrass/mqttclient/AwsIotMqtt5Client.java

```diff
@@ -0,0 +1,416 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.mqttclient;
+
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.logging.api.LogEventBuilder;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.mqttclient.v5.PubAck;
+import com.aws.greengrass.mqttclient.v5.Publish;
+import com.aws.greengrass.mqttclient.v5.Subscribe;
+import com.aws.greengrass.mqttclient.v5.SubscribeResponse;
+import com.aws.greengrass.mqttclient.v5.UnsubscribeResponse;
+import com.aws.greengrass.util.Utils;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Setter;
+import software.amazon.awssdk.crt.CRT;
+import software.amazon.awssdk.crt.mqtt5.Mqtt5Client;
+import software.amazon.awssdk.crt.mqtt5.Mqtt5ClientOptions;
+import software.amazon.awssdk.crt.mqtt5.OnAttemptingConnectReturn;
+import software.amazon.awssdk.crt.mqtt5.OnConnectionFailureReturn;
+import software.amazon.awssdk.crt.mqtt5.OnConnectionSuccessReturn;
+import software.amazon.awssdk.crt.mqtt5.OnDisconnectionReturn;
+import software.amazon.awssdk.crt.mqtt5.OnStoppedReturn;
+import software.amazon.awssdk.crt.mqtt5.PublishResult;
+import software.amazon.awssdk.crt.mqtt5.packets.ConnAckPacket;
+import software.amazon.awssdk.crt.mqtt5.packets.ConnectPacket;
+import software.amazon.awssdk.crt.mqtt5.packets.DisconnectPacket;
+import software.amazon.awssdk.crt.mqtt5.packets.PubAckPacket;
+import software.amazon.awssdk.crt.mqtt5.packets.UnsubscribePacket;
+import software.amazon.awssdk.iot.AwsIotMqtt5ClientBuilder;
+import vendored.com.google.common.util.concurrent.RateLimiter;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import javax.inject.Provider;
+
+import static com.aws.greengrass.mqttclient.AwsIotMqttClient.QOS_KEY;
+
+class AwsIotMqtt5Client implements IndividualMqttClient {
+
+    static final String TOPIC_KEY = "topic";
+    private static final String RESUB_LOG_EVENT = "resubscribe";
+    private final Provider<AwsIotMqtt5ClientBuilder> builderProvider;
+
+    private Mqtt5Client client = null;
+
+    private static final Random RANDOM = new Random();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To make your code secure, use `new SecureRandom()` to instantiate a `java.security.SecureRandom` object. Do not use `getInstance()`, `getInstanceStrong()` and `java.util.Random`.